runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/net/UrlParser.kt

```diff
@@ -25,20 +25,20 @@ internal fun urlParseImpl(url: String): Url =
 
         if (next.startsWith("/")) {
             next = next.capture(1 until next.firstIndexOrEnd("?", "#")) {
-                path = "/${it.urlDecodeComponent()}"
+                path = "/${decodingBehavior.maybeDecode(it)}"
             }
         }
 
         if (next.startsWith("?")) {
             next = next.capture(1 until next.firstIndexOrEnd("#")) {
                 it.splitAsQueryString().entries.forEach { (k, v) ->
-                    parameters.appendAll(k.urlDecodeComponent(), v.map(String::urlDecodeComponent))
+                    parameters.appendAll(decodingBehavior.maybeDecode(k), v.map { decodingBehavior.maybeDecode(it) })
```

Comment:
Question: Do we want this decode/no-decode to be a blanket setting?

The reason for shutting it off on the path is clear - but it seems like you'd always want the query decoded (and it would always be safe to do so).

Reply:
Yeah you're probably right. Will revise.